CONTRIBUTING.md

```diff
@@ -146,6 +85,414 @@ the time to ensure your contributions are high quality and easy for our
+
+### Use Descriptive Variable Names Sparingly
+
+Quoting the Go [code review comments]:
+
+> Variable names in Go should be short rather than long. This is especially true
+> for local variables with limited scope. Prefer `c` to `lineCount`. Prefer `i`
+> to `sliceIndex`.
+>
+> The basic rule: the further from its declaration that a name is used, the more
+> descriptive the name must be. For a method receiver, one or two letters is
+> sufficient. Common variables such as loop indices and readers can be a single
+> letter (`i`, `r`). More unusual things and global variables need more
+> descriptive names.
+
+Another way to frame the above is that we prefer to use short variables in all
+cases where a (human) reader could easily infer what the variable was from its
+source. For example:
+
+```go
+
+// NumberOfGeese might be used outside this package, or many many lines further
+// down the file so it needs a descriptive name. It's also just an int, which
+// doesn't give the reader much clue about what it's for.
+const NumberOfGeese = 42
+
+// w is plenty for the first argument here. Naming it gooseWrangler is redundant
+// because readers can tell what it is from its type. looseGeese on the other
+// hand warrants a descriptive name. It's short lived (lines wise), and its type
+// doesn't give us any context about what it's for.
+func capture(w goose.Wrangler, looseGeese int) error {
+        // Important goose capturing logic.
+        for looseGeese > 0 {
+                // It's not obvious from the w.Wrangle method name what the
+                // return value is, so a descriptive name names sense here too.
+                captured, err := w.Wrangle()
+                if err != nil {
+                        return errors.Wrap(err, "defeated by geese")
+                }
+                looseGeese = looseGeese - captured
+        }
+
+        // We prefer 'y' to 'yard' here because 'yard' is implied by 'NewYard'.
+        y := goose.NewYard(w)
+        return y.Secure()
+}
+```
+
+### Don't Wrap Function Signatures
+
+Quoting again from the Go [code review comments]:
+
+> Most of the time when people wrap lines "unnaturally" (in the middle of
+> function calls or function declarations, more or less, say, though some
+> exceptions are around), the wrapping would be unnecessary if they had a
+> reasonable number of parameters and reasonably short variable names. Long
+> lines seem to go with long names, and getting rid of the long names helps a
+> lot.
+
+```go
+func capture(gooseWrangler goose.Wrangler, looseGeese int, gooseYard goose.Yard,
+        duckWrangler duck.Wrangler, looseDucks, duckYard duck.Yard) error {
+        // Important fowl wrangling logic.
+}
+```
+
+If you find the need to wrap a function signature like the above it's almost
+always a sign that your argument names are superfluously verbose, or that your
+function is doing too much.
+
+If your function needs to take many optional arguments, perhaps to enable
```

Comment:
It might be worth mentioning that in the case of variadic options, wrapping for readability may make sense (e.g. instantiating a managed reconciler).

Reply:
Done. (Though TBH I have no issue with wrapping function _calls_ - it's a shame I couldn't easily gloss over that part of the quote heh.)